test/cursor.js

```diff
@@ -1,877 +1,700 @@
-          done();
-        }).error(done);
-      }
-      counter++;
-    });
+    const promise = new Promise((resolve, reject) => {
+      feed.on('error', reject)
+      feed.on('end', resolve)
+    })
 
-    yield r.db(dbName).table(tableName).insert({id: 0});
+    await feed.close()
+    await promise
+  })
 
-    //done();
-  }
-  catch(e) {
-    done(e);
-  }
-})
+  it('`on` should work on feed', async () => {
+    feed = await r.db(dbName).table(tableName2).changes().run()
+    assert(feed)
 
+    const promise = new Promise((resolve, reject) => {
+      let i = 0
+      feed.on('data', function () {
+        i++
+        if (i === smallNumDocs) {
+          feed.close().then(resolve).error(reject)
+        }
+      })
+      feed.on('error', reject)
+    })
 
-It('`next` should work on a feed', function* (done) {
-  try {
-    yield util.sleep(1000);
-    feed = yield r.db(dbName).table(tableName2).changes().run();
-    setTimeout(function() {
-      r.db(dbName).table(tableName2).update({foo: r.now()}).run();
-    }, 100)
-    assert(feed);
-    var i=0;
-    while(true) {
-      result = yield feed.next();
-      assert(result);
-      i++;
-      if (i === smallNumDocs) {
-        yield feed.close();
-        done();
-        break;
-      }
+    await r.db(dbName).table(tableName2).update({foo: r.now()}).run()
+    await promise
+  })
+
+  it('`on` should work on cursor - a `end` event shoul be eventually emitted on a cursor', async () => {
+    cursor = await r.db(dbName).table(tableName2).run({cursor: true})
+    assert(cursor)
+
+    const promise = new Promise((resolve, reject) => {
+      cursor.on('end', resolve)
+      cursor.on('error', reject)
+    })
+
+    await r.db(dbName).table(tableName2).update({foo: r.now()}).run()
+    await promise
+  })
+
+  it('`next`, `each`, `toArray` should be deactivated if the EventEmitter interface is used', async () => {
+    feed = await r.db(dbName).table(tableName2).changes().run()
+
+    feed.on('data', () => {})
+    feed.on('error', assert.fail)
+
+    try {
+      feed.next()
+      assert.fail('should throw')
+    } catch (e) {
+      assert(e.message === 'You cannot call `next` once you have bound listeners on the Feed.')
+      await feed.close()
     }
-  }
-  catch(e) {
-    done(e);
-  }
-})
-It('`next` should work on an atom feed', function* (done) {
-  try {
-    var idValue = uuid();
-    feed = yield r.db(dbName).table(tableName2).get(idValue).changes({includeInitial: true}).run();
-    setTimeout(function() {
-      r.db(dbName).table(tableName2).insert({id: idValue}).run();
-    }, 100)
-    assert(feed);
-    var i=0;
-    var change = yield feed.next();
-    assert.deepEqual(change, {new_val: null});
-    change = yield feed.next();
-    assert.deepEqual(change, {new_val: {id: idValue}, old_val: null});
-    yield feed.close();
-
-    done();
-  }
-  catch(e) {
-    done(e);
-  }
-})
+  })
+
+  it('Import with cursor as default', async () => {
```

Comment:
This test case causes the test to hang after completion. Newer versions of Mocha hangs on unresolved promises.